src/theory/quantifiers/quantifiers_attributes.cpp

```diff
@@ -200,7 +204,28 @@ void QuantAttributes::computeQuantAttributes( Node q, QAttributes& qa ){
       }
       else if (k == INST_ATTRIBUTE)
       {
-        Node avar = q[2][i][0];
+        Node avar;
+        // We support two use cases of INST_ATTRIBUTE:
+        // (1) where the user constructs a term of the form
+        // (INST_ATTRIBUTE "keyword" [node_values])
+        // (2) where we internally generate nodes of the form
+        // (INST_ATTRIBUTE v) where v has an internal attribute set on it.
+        // We distinguish these two cases by checking the kind of the first
+        // child.
+        if (q[2][i][0].getKind() == CONST_STRING)
+        {
+          // make a dummy variable to be used below
+          avar = nm->mkBoundVar(nm->booleanType());
+          std::vector<Node> nodeValues(q[2][i].begin() + 1, q[2][i].end());
+          // set user attribute on the dummy variable
+          setUserAttribute(
```

Comment:
How deep may this recurse in the worst case?

Reply:
There is no recursion here, this is called from computeQuantAttributes.

Reply:
Ah right, I missed that this was collapsed in. All good :)